src/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -964,7 +964,7 @@ private void SendMailCallback(IAsyncResult result)
                 else
                 {
                     _message.BeginSend(_writer, DeliveryMethod != SmtpDeliveryMethod.Network,
-                        ServerSupportsEai, new AsyncCallback(SendMessageCallback), result.AsyncState);
+                        IsUnicodeSupported(), new AsyncCallback(SendMessageCallback), result.AsyncState);
```

Comment:
Can you add some tests around this change please? (you can look at the tests under: corefx/src/System.Net.Mail/tests/ to get more ideas)

If adding new tests is hard, additional manual testing is ok.

Reply:
I am not familiar with writing test code.

The test needs to get the email content source code, and then analyze it to see if it works. Manual testing should be much simpler. I think so.

Before testing, we can consult the existing SmtpClient source code:
1. `ServerSupportsEai` has two references, in `IsUnicodeSupported` method and where this error occurred.
2. All `allowUnicode` parameters pass in values using `IsUnicodeSupported`, except where the error occurs, resulting in a significant difference from the `Send` method.

So, the test is half done. The rest is to send two e-mails in code for testing:
1. Use `SmtpDeliveryFormat.SevenBit`, all mail content source `subject` will be encoding.
2. Use `SmtpDeliveryFormat.International`, if the mail server supports `SMTPUTF8`, `subject` in the mail content source will be plain text, other behaviors are the same as `SevenBit` attributes.

Have a nice day~

Reply:
> Use SmtpDeliveryFormat.International, if the mail server supports SMTPUTF8, subject in the mail content source will be plain text

This is the old behavior, and the change will not affect this, right?

Reply:
> the mail content source will be plain text

Do you mean UTF-8 encoded plain text?

Reply:
Yes~

IF(`International` AND `SMTPUTF8`) Then `Plain Text`:
```
Subject: SendMailAsync ä¸­æ�� Test æµ�è¯�
```
IF(`SevenBit` OR `!SMTPUTF8`) Then `Encoded Text`: 
```
Subject: =?utf-8?B?U2VuZCDkuK3mlocgVGVzdCDmtYvor5U=?=
```